From 160df3ac40e9098942e1d2b9c7b0bd6757f4be05 Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Fri, 2 Aug 2019 07:14:11 -0400
Subject: [PATCH] Python install path: remove site-packages detection

The path `${CMAKE_INSTALL_PREFIX}/lib/pythonM.N/site-packages` is
not Debian-specific, and will exist on too many systems where it
should _not_ be assumed to be the correct install path. (This is
causing Fedora and other `lib64`-using distros to install the
bindings into `/usr/lib/`, when they were previously being
installed into the correct `/usr/lib64/` location.)
---
 src/bindings/python/CMakeLists.txt | 27 +++++++++++----------------
 1 file changed, 11 insertions(+), 16 deletions(-)

diff --git a/src/bindings/python/CMakeLists.txt b/src/bindings/python/CMakeLists.txt
index 9afabd4..80281d7 100644
--- a/src/bindings/python/CMakeLists.txt
+++ b/src/bindings/python/CMakeLists.txt
@@ -72,26 +72,21 @@ if (PYTHONLIBS_FOUND AND PYTHONINTERP_FOUND)
 	                      ${PYTHON_LIBRARIES} openshot)
 
 	### Check if the following Debian-friendly python module path exists
-	SET(PYTHON_MODULE_PATH "${CMAKE_INSTALL_PREFIX}/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages")
+	SET(PYTHON_MODULE_PATH "${CMAKE_INSTALL_PREFIX}/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/dist-packages")
 	if (NOT EXISTS ${PYTHON_MODULE_PATH})
 
-		### Check if another Debian-friendly python module path exists
-		SET(PYTHON_MODULE_PATH "${CMAKE_INSTALL_PREFIX}/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/dist-packages")
-		if (NOT EXISTS ${PYTHON_MODULE_PATH})
-
-			### Calculate the python module path (using distutils)
-			execute_process ( COMMAND ${PYTHON_EXECUTABLE} -c "\
+		### Calculate the python module path (using distutils)
+		execute_process ( COMMAND ${PYTHON_EXECUTABLE} -c "\
 from distutils.sysconfig import get_python_lib; \
 print( get_python_lib( plat_specific=True, prefix='${CMAKE_INSTALL_PREFIX}' ) )"
-				OUTPUT_VARIABLE _ABS_PYTHON_MODULE_PATH
-				OUTPUT_STRIP_TRAILING_WHITESPACE )
-
-			GET_FILENAME_COMPONENT(_ABS_PYTHON_MODULE_PATH
-					"${_ABS_PYTHON_MODULE_PATH}" ABSOLUTE)
-			FILE(RELATIVE_PATH _REL_PYTHON_MODULE_PATH
-					${CMAKE_INSTALL_PREFIX} ${_ABS_PYTHON_MODULE_PATH})
-			SET(PYTHON_MODULE_PATH ${_ABS_PYTHON_MODULE_PATH})
-		endif()
+			OUTPUT_VARIABLE _ABS_PYTHON_MODULE_PATH
+			OUTPUT_STRIP_TRAILING_WHITESPACE )
+
+		GET_FILENAME_COMPONENT(_ABS_PYTHON_MODULE_PATH
+				"${_ABS_PYTHON_MODULE_PATH}" ABSOLUTE)
+		FILE(RELATIVE_PATH _REL_PYTHON_MODULE_PATH
+				${CMAKE_INSTALL_PREFIX} ${_ABS_PYTHON_MODULE_PATH})
+		SET(PYTHON_MODULE_PATH ${_ABS_PYTHON_MODULE_PATH})
 	endif()
 	message("PYTHON_MODULE_PATH: ${PYTHON_MODULE_PATH}")
 
-- 
2.21.0

